File: tui/shared/src/scala/tui/internal/ranges.scala
package tui
package internal

object ranges {
  @inline def revRange(fromInclusive: Int, toExclusive: Int)(f: Int => Unit): Unit = {
    var idx = toExclusive - 1
    while (idx >= fromInclusive) {
      f(idx)
      idx -= 1
    }
  }

  @inline def range(fromInclusive: Int, toExclusive: Int)(f: Int => Unit): Unit = {
    var idx = fromInclusive
    while (idx < toExclusive) {
      f(idx)
      idx += 1
    }
  }
}

File: tui/src/java/tui/internal/Ranges.java
package tui.internal;

import java.util.function.IntConsumer;

public final class Ranges {
  private Ranges() {}

  public static void revRange(int fromInclusive, int toExclusive, IntConsumer f) {
    int idx = toExclusive - 1;
    while (idx >= fromInclusive) {
      f.accept(idx);
      idx -= 1;
    }
  }

  public static void range(int fromInclusive, int toExclusive, IntConsumer f) {
    int idx = fromInclusive;
    while (idx < toExclusive) {
      f.accept(idx);
      idx += 1;
    }
  }
}
